### ISSUE TITLE
Milestones for next release

### ISSUE THREAD

**Issue by jspaezp** [bittremieux/spectrum_utils#33]
Hello there!

It is me again!
I was wondering if you had a list of features in mind for the next release of the package.

I am interested in using the package for a project of mine and since I noticed there will be a fair amount of changes in the API
based on the current state of the master branch I was hoping to help that release get out of the way.

We could set up a milestone for V0.36 (maybe ...)

The prelim list I can think of is:

- Bugfix
    - [X] Empty spectrum plotting https://github.com/bittremieux/spectrum_utils/issues/20
- Tech Debt
    - [X] Refactor tests so they conform to the current status of the project
    - [X] add a .gititgnore
    - [X] ~~Add building documentation to github actions~~
- Features
    - [x] https://github.com/bittremieux/spectrum_utils/issues/31
- Documentation:
    - [X] Proforma Documentation https://github.com/bittremieux/spectrum_utils/issues/28
    - [X] Update Benchmarks
    - [X] Contributions.md addition https://github.com/bittremieux/spectrum_utils/issues/27
        - I feel like it could include the standard good stuff 
            - flake8, black, isort, pytest, tox (??) (add a pre-commit ?)
            - How to install for development ... (what the preferred workflow is)
            - What branch to PR against
            - How versioning works
            - When to create an issue vs when to create a PR
        - How to build/preview documentation - sphinx (I am not super familiar with sphinx, i would need help with this 
one), I have personally used mkdocs for all of my needs and it seems a lot easier to handle and makes a great job getting the documentation from docstrings in the code itself into the website (we could consider migrating the docs to it).
      - https://github.com/rochacbruno/python-project-template/blob/main/CONTRIBUTING.md this seems like a good template to start with ...
    - [X]  Pyteomics-cytonize documentation https://github.com/bittremieux/spectrum_utils/issues/22
    - [X] Update all the other documentation ... 
        - I recently moved my "notebook style" documentation to Quarto (https://quarto.org/) .qmd notebooks and life is better now. It allows a lot better testing of it and generates .md files with all the required outputs automatically (instead of having to update manually the theoretical outputs of the python chunks in the .md files)
        -  For example https://github.com/jspaezp/elfragmentador/blob/main/notebooks/docs/visualize_attention.qmd renders to https://jspaezp.github.io/elfragmentador/notebooks/visualize_attention/
        - If this is something you are interested in doing, it would be fairly easy based on the way you have structured the project.

So .... Do you want to start a branch to get those out of the way? or if I have a contribution to get those out of the way should I PR against master?

Best!
Sebastian

**Comment by bittremieux** [comment]
Thanks for giving me a kick under the butt! It was necessary.

And also thanks for the list of milestones. I've addressed all tasks for now and released spectrum_utils version 0.4.0. :tada:

I can still make improvements to some of these tasks. For example, the documentation now is still all static markdown pages with the code examples (although fully functional) run externally, rather than writing the relevant pages as notebooks so that the examples are automatically executed.
That doesn't change the functionality, or even the look, at the moment though. So to provide the new functionality to you and other users, I've published the new release. Documentation updates (and possibly some bug fixes as needed) will be added in patch versions.

The only milestone that was not explicitly addressed is adding documentation building to the GitHub Actions, because it's doen automatically by readthedocs.
